mixnet/client/src/lib.rs

```diff
@@ -22,21 +22,19 @@ impl MixnetClient {
     pub async fn run(config: MixnetClientConfig) -> Result<Self, Box<dyn Error>> {
         let (message_tx, _) = broadcast::channel(CHANNEL_SIZE);
 
-        Receiver::run(config.listen_addr, message_tx.clone()).await?;
+        // Run a receiver only if listen_address is specified
+        if let Some(listen_address) = config.listen_address {
+            Receiver::run(listen_address, message_tx.clone()).await?;
+        }
```

Comment:
This can be a method over the `MixnetMode` where run does nothing in the `SenderMode`

Reply:
Thank you.
I defined the `MixnetClientMode` enum, as you suggested. But, instead of defining methods over it, I refactored the `MixnetClient` initialization in a different way: e895c26, because it was not easy for me to move all logic into `MixnetClientMode`, than I expected.
Please let me know if you see any better way.

Reply:
This discussion has been moved to https://github.com/logos-co/nomos-node/commit/e895c26564e2c49c97b53eb5070603911cad0c33#r124833675